crates/ingress-http/src/handler/service_handler.rs

```diff
@@ -161,7 +160,12 @@ where
             let mut service_invocation =
                 ServiceInvocation::initialize(invocation_id, invocation_target, Source::Ingress);
             service_invocation.with_related_span(SpanRelation::Parent(ingress_span_context));
-            service_invocation.idempotency = idempotency;
+            if let Some(key) = idempotency_key {
+                service_invocation.idempotency_key = Some(key);
+                // TODO set this value even in non-idempotency cases
+                service_invocation.completion_retention_time =
+                    Some(invocation_target_meta.idempotency_retention);
```

Comment:
I guess this would be needed for the workflow results to be kept around, right?

Reply:
yep. a later commit should have fixed this.